### ISSUE TITLE
Download list doesn't survive a restart

### ISSUE THREAD

**Issue by thany** [downthemall/downthemall#486]
**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser and version: Firefox 115.5 ESR
 - DownThemAll! version: 4.11.1

**Describe the bug**
When the browser needs to restart for any reason whatsoever, I would expect the list of pending downloads to survive that. DTA should not forget my meticulously prepared list of downloads like it was nothing. This is a form of data loss, and I feel this deserves elevated priority.

**To Reproduce**
1. Queue something to download
2. Restart the browser
3. See how empty the queue is

**Expected behavior**
The *entire* queue, with all meta information like filenames and destination directories, all paused, completed, and pending downloads, should be preserved after a restart.

**Additional context**
As a workaround, I thought export would work. For some reason this addon has 3 methods of export, none of which also export the destination directory, and frankly to me this is baffling. At least it can import its own exports, but they are incomplete.

There is currently no way to preserve a queue with ALL information, other than to keep Firefox running until it's done. Sometimes this can take days, or a host may be temporarily offline, and then what?

**Comment by nmaier** [comment]
DownThemAll! does persist the data. However, I believe if you put Firefox into permanent private browsing mode, then the browser itself decides not to actually persist anything, not even databases from extensions, such as the one DTA uses. Do you happen to use permanent private browsing mode? Or any other preferences or tools that might clear out all "site data" upon restart (extensions are threated like special websites by the browser)?

If not, there might be some kind of data corruption. But It's rather unlikely that everything get corrupted, unless the entire physical data file the browser stores on disk is somehow corrupted...

**Comment by thany** [comment]
Yes, permanent private mode might be the cause. Do you happen to know if there's a way around it to except this addon from getting its data cleared?
